tests/v2/utils/test_utils.go

```diff
@@ -55,14 +60,23 @@ func init() {
 		}
 		testLogger.Println("Test Starting:")
 	}
+
 }
 
-// createTempDir creates a specified sub directory under the temp directory if it does not exist.
+// CreateTempDir creates a specified sub directory under the temp directory if it does not exist.
 // Returns the name of the created directory.
-func createTempDir(subdir string) string {
+func CreateTempDir(subdir string) string {
 	tempDir := tmpDir + subdir + "/"
 	if _, err := os.Stat(tempDir); os.IsNotExist(err) {
-		os.Mkdir(tempDir, 0755)
+		if err = os.Mkdir(tempDir, 0755); err != nil {
+			LogErrorMessage(fmt.Sprintf("Failed to create temp directory %s will use %s : %v", tempDir, tmpDir, err))
+			// if cannot create subdirectory just use the base tmp directory
+			tempDir = tmpDir
+		}
+	} else if err != nil {
```

Comment:
why we need this extra else statement?
can pre-define the error

```
var err error
	if _, err = os.Stat(tempDir); os.IsNotExist(err) {
		err = os.Mkdir(tempDir, 0755)
	}
 if err != nil {
			LogErrorMessage(fmt.Sprintf("Failed to create temp directory %s will use %s : %v", tempDir, tmpDir, err))
			// if cannot create subdirectory just use the base tmp directory
			tempDir = tmpDi
```

Reply:
with your example if os.IsNotExist(err) returned true your second if would also return true which would be a bug.  The else statement deals with errors which are not the  os.IsNotExist(err) for example a permission error.

Reply:
The second won't return true if `os.Mkdir`  passes, err will be nil in this case

Reply:
ah you removed my logic processing the os.Mkdir error, see where you are coming from now. Let me update it.

Reply:
Uodated